includes/class-wcmp-frontend.php

```diff
@@ -11,548 +11,658 @@
+            }
+
+            if ( isset( $_POST['post_data'] ) ) {
+                // non-default post data for AJAX calls
+                parse_str( $_POST['post_data'], $post_data );
+
+                return $post_data;
+            }
+
+            // checkout finalization
+            return $_POST;
+        }
+
+        private function add_fee( $fee_name, $fee ) {
+            $fee = $this->normalize_price( $fee );
+            // get shipping tax data
+            $shipping_tax_class = $this->get_shipping_tax_class();
+            if ( $shipping_tax_class ) {
+                if ($shipping_tax_class == 'standard') {
+                    $shipping_tax_class = '';
+                }
+                WC()->cart->add_fee( $fee_name, $fee, true, $shipping_tax_class );
+            } else {
+                WC()->cart->add_fee( $fee_name, $fee );
+            }
+        }
+
+        /**
+         *
+         * Check witch delivery option is selected
+         *
+         * @param $post_data
+         * @param $post_data_value
+         * @param $delivery_type
+         * @param $backend_setting
+         * @param $delivery_titel
+         *
+         * @return bool
+         */
+        private function add_fee_from_setting( $post_data, $post_data_value ,$delivery_type, $backend_setting, $delivery_titel ) {
+            // Fee for "delivery" option
+            if (isset($post_data[$post_data_value]) && $post_data[$post_data_value] == $delivery_type) {
+                if ( ! empty( WooCommerce_MyParcel()->checkout_settings[$backend_setting] ) ) {
+                    $fee      = WooCommerce_MyParcel()->checkout_settings[$backend_setting];
+                    $fee_name = __( $delivery_titel, 'woocommerce-myparcel' );
+                    $this->add_fee( $fee_name, $fee );
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private function add_fee_signature( $delivery_options, $delivery_titel ) {
+            if ( $delivery_options['signed'] !== 1){
+                return;
+            }
+
+            $fee      = WooCommerce_MyParcel()->checkout_settings['signed_fee'];
+
+            if ( ! empty( $fee ) ) {
+                $fee_name = __( $delivery_titel, 'woocommerce-myparcel' );
+                $this->add_fee( $fee_name, $fee );
+            }
+        }
+
+        private function add_fee_only_recipient( $delivery_options, $delivery_titel ) {
+            if ( $delivery_options['only_recipient'] !== 1){
+                return;
+            }
+
+            $fee      = WooCommerce_MyParcel()->checkout_settings['only_recipient_fee'];
+
+            if ( ! empty( $fee ) ) {
+                $fee_name = __( $delivery_titel, 'woocommerce-myparcel' );
+                $this->add_fee( $fee_name, $fee );
+            }
+        }
+
+        /**
+         * @return string
+         */
+        private function getDeliveryOptionsShippingMethods() {
```

Comment:
getDeliveryOptionsShippingMethods moet (alleen in wp) snake case zijn

Reply:
Done!